pkg/controller/operands/kubevirt_test.go

```diff
@@ -247,6 +257,205 @@ var _ = Describe("KubeVirt Operand", func() {
 			Expect(foundResource.Data).To(Equal(outdatedResource.Data))
 			Expect(foundResource.Data).To(Not(Equal(expectedResource.Data)))
 		})
+
+		Context("Feature Gates", func() {
+			origKvFeatureGates := managedKvFeatureGates
+
+			const (
+				constFeatureGates = "DataVolumes,SRIOV,LiveMigration,CPUManager,CPUNodeDiscovery,Sidecar,Snapshot"
+				userModifiedFgs   = "userModifiedFg1,userModifiedFg2,userModifiedFg3"
+			)
+			BeforeEach(func() {
+				// list of fake KV managed FGs
+				managedKvFeatureGates = []string{fgEnabled, fgMissing, fgDisabled, fgNoChange}
+			})
+
+			AfterEach(func() {
+				managedKvFeatureGates = origKvFeatureGates
+			})
+
+			It("should not add the feature gates if they are not set in HyperConverged CR", func() {
+				existingResource := NewKubeVirtConfigForCR(hco, commonTestUtils.Namespace)
+				By("KV CR should contain the fgEnabled feature gate", func() {
+					Expect(existingResource.Data[virtconfig.FeatureGatesKey]).Should(Equal(constFeatureGates))
+				})
+			})
+
+			It("should add the feature gates if they are set in HyperConverged CR", func() {
```

Comment:
maybe for clarity we can rephrase to `if they exist and enabled..` just when we say `set` it can be also set to false and then it won't appear in the list